docs/04-reference/wingsdk-spec.md

```diff
@@ -253,105 +253,115 @@ resource Bucket {
   /**
    * Run an inflight whenever an object is uploaded to the bucket.
    */
-  onCreate(fn: inflight (key: str) => void, opts: BucketOnUploadProps?): void;
+  onCreate(fn: IBucketEventHandler, opts?: BucketOnUploadProps): void;
 
   /**
    * Run an inflight whenever an object is deleted from the bucket.
    */
-  onDelete(fn: inflight (key: str) => void, opts: BucketOnDeleteProps?): void;
+  onDelete(fn: IBucketEventHandler, opts?: BucketOnDeleteProps): void;
 
   /**
    * Run an inflight whenever an object is updated in the bucket.
    */
-  onUpdate(fn: inflight (key: str) => void, opts: BucketOnUpdateProps?): void;
+  onUpdate(fn: IBucketEventHandler, opts?: BucketOnUpdateProps): void;
 
   /**
    * Run an inflight whenever an object is uploaded, modified, or deleted from the bucket.
    */
-  onEvent(fn: inflight (event: BucketEvent) => void, opts: BucketOnEventProps?): void;
+  onEvent(fn: IBucketEventHandler, opts?: BucketOnEventProps?): void;
 
   /**
    * Add an object to the bucket that is uploaded when the app is deployed.
    */
-  addObject(key: str, value: Blob): void;
+  addObject(key: string, body: string): void;
 
   /**
    * Upload an object to the bucket.
    */
-  inflight put(key: str, value: Blob): void;
+  inflight put(key: string, body: string): Promise<void>;
```

Comment:
All inflight methods are async in Wing by default, so the return type does not need to be changed here. (if you mouse over the word "put" when you have some Wing code open in your favorite IDE, we want the return type to just show `void` and not `Promise<void>`